### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by thanish**
My first xgb attempt. The code is present in XGB_sub_1.R and the predicted facies in XGB_predicted_facies_1.csv

### CODE REVIEW

**Comment by kwinkunks** [comment]
Hry Thanish... that CSV scores *0.520*. Not too bad... could be a good start for some tuning.

**Comment by thanish** [comment]
Ahhhh that feels very bad for me. My local validation accuracy scores high, but the submission scores don't go up. I'm using accuracy and not F1 in my evaluation. But still I believe more the accuracy of the confusion matrix more should be the F1. All the searches I did only shows me F1 for binary class for R and most places they don't explain if there is a F1 function for multiclass in R. 

image

**Comment by kwinkunks** [comment]
Are your local accuracies measured by dropping out entire wells, or only random subsets of the data? The first approach, perhaps even dropping pairs of wells, would be the closest thing to the contest conditions.

**Comment by thanish** [comment]
Hi @kwinkunks all my accuracy/validation is only random split among the train dataset into train and validation dataset. By pairs if well you mean the blind dataset("CRAWFORD","STUART") I have used them only to predict when creating the submission file. I hope I'm within the competition condition. Just to add on there is no difference between the my approach in SVM_1, RF_1 and XGB_1 except the different algorithms used.  But you can see the difference in validation accuracy between the them, but submission F1 hardly increased. So I was blinking what am I doing wrong.

**Comment by kwinkunks** [comment]
The problem with a random split is that you are then training on data whose samples are immediately adjacent to samples you are testing on. Since there is at least some spatial correlation (one sample tends to look a bit like the next sample), you get accuracy that is much higher than you would get if you split on a group. To put it another way: the samples are not independent.

You can't include the blind wells in your validation, because you don't know their labels. But you could make your 'test' group by leaving out entire wells, or pairs of wells... your accuracy will go down and be much closer to what you're going to get in validation.

There's an example of this approach (using scikit-learn in Python) here > https://github.com/seg/2016-ml-contest/blob/master/LiamLearn/K-fold_CV_F1_score__MATT.ipynb 

I expect there's something similar in R.

**Comment by thanish** [comment]
Okay @kwinkunks got it. I think the split was the problem which was raising my hopes high. I will go on with my validation dataset to be a complete well or may be two. Moreover I now have the function to directly use F1 as my evaluation metric rather than using accuracy. So I should be able to compare better with my validation and what it scores on the blind set. Thanks for your help :)
